### PR TITLE
FEAT: Generate GTFS Schedule data for Bus Performance Application

### PR DESCRIPTION
**Pull request by rymarczy**
This change adds functionality to Bus Performance Manager to generate any/all data needed from the GTFS static schedule. 

All data generation is based on `service_date` and is sourced from the Compressed GTFS Archive files. Local version of the GTFS Archive files are cached by `service_date`, so any new service date will trigger the re-downloading of GTFS files (hopefully once per day). 

I also updated Polars to version 1.2, which required a few changes to existing function calls in other files. 

It is likely that the export dataframe will required some re-naming in the future, depending on how this data is joined to realtime records. 

Asana Task: https://app.asana.com/0/1205827492903547/1207771349226050

### CODE REVIEW

**Comment by rymarczy** [comment]
I held off on re-basing this onto your new testing framework, I can roll that into a future commit when #410 has been merged to main.

**Comment by rymarczy** [comment]
Locally, this spits out schedule data for a service_date in about 1 second (not counting parquet file downloading), which is a significant improvement compared to the RDS based system.

File: tests/bus_performance_manager/bus_test_gtfs.csv
Comment (rymarczy) [inline tests/bus_performance_manager/bus_test_gtfs.csv:1]: This is 20 randomly selected trips from GTFS data generated fro Aug 1, 2024. Can adjust as we get review comments from Ops Analytics.

**Comment by rymarczy** [comment]
Testing ended up being pretty self contained, so just went ahead and pushed it up to this branch.

**Review by mzappitello — COMMENTED** [review]
A few questions / suggestions, but mostly LGTM

File: src/lamp_py/bus_performance_manager/gtfs.py [hunk 1]
```diff
@@ -0,0 +1,417 @@
...
+        plan_route_direction_headway_seconds -> Int64
+        plan_direction_destination_headway_seconds -> Int64
+    """
+    sync_gtfs_files(service_date)
+
+    stop_events = stop_events_for_date(service_date)
+
+    stop_events = stop_event_metrics(stop_events)
+
+    return stop_events
```
[the remark this answers is not in the document]
Comment (rymarczy) [reply src/lamp_py/bus_performance_manager/gtfs.py:417]: I don't think it's required, for now. Running locally (with synced source files) generates the dataset in less than a second. If that duration ends up taking longer we can look into caching.

File: tests/bus_performance_manager/test_gtfs_rt_ingestion.py
Comment (mzappitello) [inline tests/bus_performance_manager/test_gtfs_rt_ingestion.py:1]: woops.

File: src/lamp_py/bus_performance_manager/gtfs.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (mzappitello) [reply src/lamp_py/bus_performance_manager/gtfs.py:429]: so speedy these days.

**Review by mzappitello — APPROVED** [review]
🍰

**Pull request merged**